test/gutter_tree_test.cpp

```diff
@@ -14,87 +16,88 @@ static std::atomic<uint32_t> upd_processed;
 // queries the buffer tree and verifies that the data
 // returned makes sense
 // Should be run in a seperate thread
-void querier(BufferTree *buf_tree, int nodes) {
+void querier(GutterTree *gt, int nodes) {
   data_ret_t data;
   while(true) {
-    bool valid = buf_tree->get_data(data);
+    bool valid = gt->get_data(data);
     if (valid) {
       Node key = data.first;
       std::vector<Node> updates = data.second;
       // verify that the updates are all between the correct nodes
       for (Node upd : updates) {
         // printf("edge to %d\n", upd.first);
         ASSERT_EQ(nodes - (key + 1), upd) << "key " << key;
-        upd_processed += 1;
+        upd_processed++;
```

Comment:
if you really want to be pedantic, it should probably be `++upd_processed`

Reply:
Why is that? Shouldn't they have the same result?

Reply:
Yes. Again it's a very nitpicky thing; I'm sure you're aware of `++i` vs `i++`. In C++ `++i` is preferred but it doesn't matter too much either way.

Reply:
Just looked this up. That’s really interesting. I always assumed the compiler would just optimize out the difference

Reply:
This matters not at all for primitives but can matter a lot if you're declaring these operators yourself. Just a fun fact I guess.